### PR TITLE
Update java6 to install also on Catalina

### PR DESCRIPTION
**Pull request by nicerloop**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/java6.rb [hunk 1]
```diff
@@ -7,11 +7,12 @@
   name 'Apple Java 6 Standard Edition Development Kit'
   homepage 'https://support.apple.com/kb/DL1572'
 
-  pkg 'JavaForOSX.pkg'
+  container nested: 'JavaForOSX.pkg'
 
-  uninstall pkgutil: [
-                       'com.apple.pkg.JavaForMacOSX107',
-                       'com.apple.pkg.JavaMDNS',
-                       'com.apple.pkg.JavaEssentials',
```
Comment (reitermarkus) [inline Casks/java6.rb]: Does this also handle the last two?

File: Casks/java6.rb [hunk 2]
```diff
@@ -7,11 +7,12 @@
...
-                     ]
+  artifact 'Library/Java/JavaVirtualMachines/1.6.0.jdk', target: '/Library/Java/JavaVirtualMachines/1.6.0.jdk'
+
+  preflight do
+    `cd "#{staged_path}" && pkgutil --expand JavaForOSX.pkg JavaForOSX`
```
Comment (reitermarkus) [inline Casks/java6.rb]: Use `system_command`.

File: Casks/java6.rb (hunk 1 above)
Comment (nicerloop) [reply Casks/java6.rb]: These packages do not appear to be registered on my Mojave machine:

$ brew cask uninstall java6
==> Uninstalling Cask java6
==> Uninstalling packages:
com.apple.pkg.JavaForMacOSX107
Unknown error Error Domain=NSCocoaErrorDomain Code=4 "“com.apple.pkg.JavaForMacOSX107.bom” couldn’t be removed." UserInfo={NSFilePath=/var/db/receipts/com.apple.pkg.JavaForMacOSX107.bom, NSUserStringVariant=(
    Remove
), NSUnderlyingError=0x7fd89f415fa0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}.
==> Purging files for version 1.6.0_65-b14-468 of Cask java6

I have manually searched the pkg and only found a reference to com.apple.pkg.JavaForMacOSX107 which only installs the 1.6.0.jdk artifact.

File: Casks/java6.rb (hunk 2 above)
Comment (nicerloop) [reply Casks/java6.rb]: Sorry, but I have no experience with ruby other than some copy/paste for some casks maintenance. Could you give me any pointer for the desired modification?

Comment (reitermarkus) [reply Casks/java6.rb]: Look at some other casks using `system_command`, e.g. https://github.com/Homebrew/homebrew-cask/blob/1c980c8b02a3e2deb9d593edf29f15c732c43e49/Casks/pokemon-trading-card-game-online.rb#L14-L15.

You will also need to add the `chdir:` parameter to change the directory.

Comment (nicerloop) [reply Casks/java6.rb]: I found some examples through the repository, made changes that seem to work

File: Casks/java6.rb [hunk 3]
```diff
@@ -18,11 +18,10 @@
   else
     container nested: 'JavaForOSX.pkg'
```
Comment (reitermarkus) [inline Casks/java6.rb]: Why is this necessary?

Comment (nicerloop) [reply Casks/java6.rb]: It extracts the pkg from the dmg

Comment (nicerloop) [reply Casks/java6.rb]: In fact is is not necessary.

File: Casks/java6.rb (hunk 1 above)
Comment (reitermarkus) [reply Casks/java6.rb]: Since these don't seem to be used anymore, remove them.

File: Casks/java6.rb [hunk 4]
```diff
@@ -7,11 +7,15 @@
...
-                       'com.apple.pkg.JavaForMacOSX107',
-                       'com.apple.pkg.JavaMDNS',
-                       'com.apple.pkg.JavaEssentials',
-                     ]
+    uninstall pkgutil: 'com.apple.pkg.JavaForMacOSX107'
+  else
+    artifact 'JavaForOSX/JavaForOSX.pkg/Payload/Library/Java/JavaVirtualMachines/1.6.0.jdk', target: '/Library/Java/JavaVirtualMachines/1.6.0.jdk'
+
+    preflight do
+      system_command 'pkgutil', chdir: staged_path, args: ['--expand-full', 'JavaForOSX.pkg', 'JavaForOSX']
```
Comment (vitorgalvao) [inline Casks/java6.rb]: Why is this necessary?

Comment (nicerloop) [reply Casks/java6.rb]: The pkg would not install on my Catalina machine, and I'm not the only one to face this case. In fact the pkg just install the jdk bundle in the right location: /Library/Java/JavaVirtualMachines/. Options are 1) patch the pkg Distribution script or 2) expand the pkg and move the jdk bundle to the right location. As both need expanding the pkg, I just went with the second one. Just to be safe, I set this up only for Catalina, but should be applicable to any platform.
